osu.Game/Screens/Play/HitErrorDisplay/DefaultHitErrorDisplay.cs

```diff
@@ -0,0 +1,178 @@
+                    Height = (float)((maxHitWindows - HitWindows.Good) / (maxHitWindows * 2))
+                });
+
+            bar.AddRange(new Drawable[]
+            {
+                topGreenBox = new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = colours.Green,
+                    Height = (float)((HitWindows.Good - HitWindows.Great) / (maxHitWindows * 2))
+                },
+                new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = colours.BlueLight,
+                    Height = (float)(HitWindows.Great / maxHitWindows)
+                },
+                bottomGreenBox = new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = colours.Green,
+                    Height = (float)((HitWindows.Good - HitWindows.Great) / (maxHitWindows * 2))
+                }
+            });
+
+            if (HitWindows.Meh != 0)
+                bar.Add(new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = ColourInfo.GradientVertical(colours.Yellow, colours.Yellow.Opacity(0)),
+                    Height = (float)((maxHitWindows - HitWindows.Good) / (maxHitWindows * 2))
+                });
+
+            if (HitWindows.Meh == 0)
+            {
+                topGreenBox.Colour = ColourInfo.GradientVertical(colours.Green.Opacity(0), colours.Green);
+                bottomGreenBox.Colour = ColourInfo.GradientVertical(colours.Green, colours.Green.Opacity(0));
+            }
+        }
+
+        public override void OnNewJudgement(JudgementResult newJudgement)
+        {
+            if (!newJudgement.IsHit)
+                return;
+
+            var judgementLine = CreateJudgementLine(newJudgement);
+
+            judgementsContainer.Add(judgementLine);
+
+            judgementLine.FadeOut(judgement_life_time, Easing.OutQuint).Expire();
+
+            arrow.MoveToY(calculateArrowPosition(newJudgement), arrow_move_duration, Easing.OutQuint);
+        }
+
+        protected virtual Container CreateJudgementLine(JudgementResult judgement) => new CircularContainer
+        {
+            Anchor = Anchor.Centre,
+            Origin = Anchor.Centre,
+            Masking = true,
+            RelativeSizeAxes = Axes.X,
+            Height = 2,
+            RelativePositionAxes = Axes.Y,
+            Y = getRelativeJudgementPosition(judgement.TimeOffset),
+            Child = new Box
+            {
+                RelativeSizeAxes = Axes.Both,
+                Colour = Color4.White,
+            }
+        };
+
+        private float getRelativeJudgementPosition(double value) => (float)(value / maxHitWindows);
+
+        private float calculateArrowPosition(JudgementResult newJudgement)
+        {
+            if (judgementOffsets.Count > stored_judgements_amount)
```

Comment:
is there a reason you're using a queue, rather than just a moving average?

Reply:
No, it's just a first idea I came up with

Reply:
please switch to moving average. it will feel better.